src/data/option/basic.lean

```diff
@@ -142,4 +142,25 @@ theorem lift_or_get_choice {f : α → α → α} (h : ∀ a b, f a b = a ∨ f
 | none     (some b) := or.inr rfl
 | (some a) (some b) := by simpa [lift_or_get] using h a b
 
+/-! Add functorial coercions and basic lemmas about it. -/
+section coercions
+variable [has_coe α β]
+
+instance option.to_option : has_coe (option α) (option β) := ⟨option.map (λ a, (a : β))⟩
```

Comment:
@digama0 Is this coercion safe? Or does it blow up type class search?

Reply:
It doesn't blow up typeclass search, but I'm not very excited to see more complicated coercions in general, as they tend to obfuscate proofs (because they can't be ignored in proofs but are not visible in statements).

Reply:
It doesn't blow up typeclass search, but I'm not very excited to see more complicated coercions in general, as they tend to obfuscate proofs (because they can't be ignored in proofs but are not visible in statements).

Reply:
So all in all, is there anything from this PR that should be put into mathlib? In a future submission of mine (continued fractions), I will have to coerce things like `option rat` to `option real` etc.

Reply:
Alright, I'll make it a local instance then! Cheers.